pkg/pbflow/proto.go

```diff
@@ -90,10 +90,27 @@ func FlowToPB(fr *model.Record, s *ovnobserv.SampleDecoder) *Record {
 	}
 	pbflowRecord.DupList = make([]*DupMapEntry, 0)
 	for _, intf := range fr.Interfaces {
-		pbflowRecord.DupList = append(pbflowRecord.DupList, &DupMapEntry{
+		entry := DupMapEntry{
 			Interface: intf.Interface,
 			Direction: Direction(intf.Direction),
-		})
+			Udn:       "",
+		}
+		if s != nil {
+			m, err := s.GetInterfaceUDNs()
```

Comment:
Looking at the ovn code, it seems like `s.GetInterfaceUDNs()` doesn't just returned some cached thing, it calls `d.ovsdbClient.List(context.Background(), &ifaces)` under the hood, which may involve some i/o ? That would be nice to double check. I think it's better _at the very least_ to move this call out of the interface for loop, but even better, it could be fetched just once for the whole eviction loop ?

Reply:
ovsdb is a unix socket to ovs DB `/var/run/openvswitch/db.sock` and I believe its cache based too, doing once per evicttion won't work as we might miss tags as pods comes and goes

Reply:
even if there's a cache doesn't mean there will never be i/o - it sounds risky to have that in the agent data path
... checking with the ovn folks what it really is about